diff -urN vice-2.4/src/gfxoutputdrv/gifdrv.c vice-2.4-patched/src/gfxoutputdrv/gifdrv.c
--- vice-2.4/src/gfxoutputdrv/gifdrv.c	2012-07-26 10:46:05.000000000 +1100
+++ vice-2.4-patched/src/gfxoutputdrv/gifdrv.c	2014-05-29 21:58:48.880691892 +1100
@@ -114,7 +114,11 @@
   if (EGifPutScreenDesc(sdata->fd, screenshot->width, screenshot->height, 8, 0, gif_colors) == GIF_ERROR ||
       EGifPutImageDesc(sdata->fd, 0, 0, screenshot->width, screenshot->height, 0, NULL) == GIF_ERROR)
   {
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+    EGifCloseFile(sdata->fd, NULL);
+#else
     EGifCloseFile(sdata->fd);
+#endif
     VICE_FreeMapObject(gif_colors);
     lib_free(sdata->data);
     lib_free(sdata->ext_filename);
@@ -145,7 +149,11 @@
 
     sdata = screenshot->gfxoutputdrv_data;
 
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+    EGifCloseFile(sdata->fd, NULL);
+#else
     EGifCloseFile(sdata->fd);
+#endif
     VICE_FreeMapObject(gif_colors);
 
     /* for some reason giflib will create a file with unexpected
@@ -184,7 +192,11 @@
 
 static int gifdrv_close_memmap(void)
 {
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+  EGifCloseFile(gifdrv_memmap_fd, NULL);
+#else
   EGifCloseFile(gifdrv_memmap_fd);
+#endif
   VICE_FreeMapObject(gif_colors);
   lib_free(gifdrv_memmap_ext_filename);
 
@@ -231,7 +243,11 @@
   if (EGifPutScreenDesc(gifdrv_memmap_fd, x_size, y_size, 8, 0, gif_colors) == GIF_ERROR ||
       EGifPutImageDesc(gifdrv_memmap_fd, 0, 0, x_size, y_size, 0, NULL) == GIF_ERROR)
   {
+#if GIFLIB_MAJOR * 100 + GIFLIB_MINOR >= 501
+    EGifCloseFile(gifdrv_memmap_fd, NULL);
+#else
     EGifCloseFile(gifdrv_memmap_fd);
+#endif
     VICE_FreeMapObject(gif_colors);
     lib_free(gifdrv_memmap_ext_filename);
     return -1;
